### PR TITLE
fix: problem when deleting capture areas

### LINKED ISSUE #186: Bug when having multiple Capture Areas, and deleting one out of order
# Issue description
*Describe your issue in detail here*

Noticed that if you have like 3 Capture areas and you delete nr. 2, then nr. 1 and then nr. 0, everything is fine, but if you do a different order like deleting nr.0 first you will find errors in the console.

image

And when this is done, the Order is kinda fucked and problems do occur in other ways.

# Definition Of Ready (DoR)
*This issue can be worked on if*
- [x] a developer is assigned
- [x] the issue is labeled
- [x] the issue is assigned to the current sprint
- [x] the issue is estimated
- [x] a meaningful title has been set
- [x] the issue is described in detail

### PR DESCRIPTION
**Pull request by tonoizer**
# PR description

There was a bug with the capture area array not really updating the rendered list properly because its not reactive etc. The solution may not be really good tho.

# Definition Of Done (DoD)

_This PR can be squashed / merged if_

-   [x] a developer is assigned
-   [x] the PR is NOT estimated
-   [x] the PR is labeled
-   [x] the PR is NOT assigned to the current sprint
-   [x] a meaningful title has been set according to https://www.conventionalcommits.org/
-   [x] the PR is described in detail
-   [x] the PR links to an issue
-   [x] the PR has been reviewed

_Add additional conditions here if necessary for this PR_

fix: #186

### CODE REVIEW

**Review by roman533 — DISMISSED** [review]
Simple solution, works

**Review by Claiyc — CHANGES_REQUESTED** [review]
This is actually a bigger issue than it seems… We cannot just reassign ca ids, as the VIGAD-API redis db structure relies on them (and therefore potential third party apps as well). A solution may be to have unique ids for capture areas (randomly generated). However for this solution we should provide the ability for a user to manually override the id of a capture area to keep consistency between sessions.

**Comment by BeierKevin** [comment]
First attemt to fix the issue, to add the possibility to change the CaptureArea ID can be added afterwards

**Comment by Claiyc** [comment]
> If there should be the option that the user can change the ID of a capture area i need to change a lot ofthings for example should the id then be only a number and what rules should this id follow if any ?

I guess it will be easier if it is alphanumeric (so we definitely won't run into issues with collisions). However it needs to be URI conform as well. Other than that, no rules that I can think of. Maybe not too long (idk, 6 characters should be enough?)

**Comment by BeierKevin** [comment]
If this only alphanumeric isn't it always URI conform ?

File: src/components/capture-area/CaptureAreaSearchValue/CaptureAreaSearchValue.vue [hunk 1]
```diff
@@ -56,59 +98,165 @@ import {
...
+const captureAreaIdRules = {
+    required: (value: string) =>
+        !!value || 'An access token is required to start a session',
+    uppercase: (v: string) =>
+        /[A-Z]/.test(v) || 'Must include at least one uppercase letter',
+    lowercase: (v: string) =>
+        /[a-z]/.test(v) || 'Must include at least one lowercase letter',
+    number: (v: string) =>
+        /[0-9]+/.test(v) || 'Must include at least one number',
+    max: (v: string) =>
```
Comment (Claiyc) [inline src/components/capture-area/CaptureAreaSearchValue/CaptureAreaSearchValue.vue:132]: image
does not seem to apply

File: src/components/capture-area/CaptureAreaSearchValue/CaptureAreaSearchValue.vue [hunk 2]
```diff
@@ -56,59 +98,165 @@ import {
...
 /**
  * Gives a notficiation after deletion of capture area
  * @param captureAreaId
  */
-function deleteCaptureAreaNotification(id: number): void {
+function deleteCaptureArea(id: string): void {
     useNotificationSystem().createWarningNotification({
         title: 'Capture Area Deleted',
-        message: 'The Capture Area ' + id + ' got deleted',
+        message: 'Reorganizing capture areas...',
```
Comment (Claiyc) [inline src/components/capture-area/CaptureAreaSearchValue/CaptureAreaSearchValue.vue:256]:
```suggestion
        message: 'Cleaning up...',
```

File: src/composables/useTokenGenerator/useTokenGenerator.ts [hunk 3]
```diff
@@ -66,23 +74,59 @@ export default function useTokenGenerator() {
...
-    const generateValidToken = (): string => {
-        const token = generateToken();
-        if (
-            defaultRules.lowercase(token) === true &&
-            defaultRules.uppercase(token) === true &&
-            defaultRules.special(token) === true &&
-            defaultRules.number(token) === true &&
-            defaultRules.min(token) === true
-        ) {
+    const generateValidToken = (
```
Comment (Claiyc) [inline src/composables/useTokenGenerator/useTokenGenerator.ts:86]: I guess here lies the issue regarding upper comment. length is not being checked for

File: src/proc/Vigad.ts [hunk 4]
```diff
@@ -52,43 +53,136 @@ export class Vigad {
...
+        return this.captureAreas.some((area) => area.getId() === caId);
+    }
+
+    /**
+     * Renames a capture area by updating its ID.
+     * @param {string} oldId - The old ID of the capture area.
+     * @param {string} newId - The new ID to assign to the capture area.
+     * @returns {boolean} - Returns true if the renaming was successful, false otherwise.
+     */
+    public renameCaptureArea(oldId: string, newId: string): boolean {
```
Comment (Claiyc) [inline src/proc/Vigad.ts:122]: Some weird stuff is happening here.
Suppose you enter abJk5 (valid id).
Then use arrow keys to go back 2 characters
Then click backspace and **immidiately** another character
The character gets inserted before some internal validation / transformation runs and therefore gets appended to the end of the string

File: src/proc/Vigad.ts [hunk 5]
```diff
@@ -52,43 +53,136 @@ export class Vigad {
...
+        // Return true to indicate successful renaming
+        return true;
+    }
+
+    /**
+     * Delete the capture area with the specified ID.
+     * @param {number} id - The ID of the capture area to delete.
+     * @returns {boolean} - Returns true if the capture area is successfully deleted, false otherwise.
+     */
+    public deleteCaptureArea(id: string): boolean {
```
Comment (Claiyc) [inline src/proc/Vigad.ts:156]: deleting a capture area does not remove the id from upload queue:
image

**Review by Claiyc — CHANGES_REQUESTED** [review]

File: src/components/capture-area/CaptureAreaSearchValue/CaptureAreaSearchValue.vue (hunk 1 above)
Comment (BeierKevin) [reply src/components/capture-area/CaptureAreaSearchValue/CaptureAreaSearchValue.vue:132]: What do you mean, what does not apply ?

Comment (Claiyc) [reply src/components/capture-area/CaptureAreaSearchValue/CaptureAreaSearchValue.vue:132]: max length

Comment (BeierKevin) [reply src/components/capture-area/CaptureAreaSearchValue/CaptureAreaSearchValue.vue:132]: Yeah max lenght of 6 chars you only have 3 there

File: src/proc/Vigad.ts (hunk 4 above)
Comment (BeierKevin) [reply src/proc/Vigad.ts:122]: Hmm didn't noticed that but for some reason I had problems with this function before that it unfocused the input field when the id was valid which was weird as well and I could explain myself why

Comment (BeierKevin) [reply src/proc/Vigad.ts:125]: I am very confused why the input field in general is at all unselected, can not figure out why this is the case at all

**Comment by BeierKevin** [comment]
This is just a workaround, because i cant fix it really

**Review by Claiyc — APPROVED** [review]
seems to get the job done

**Pull request merged**